tests/suites/test_suite_x509write.function

```diff
@@ -39,25 +39,29 @@ static int x509_crt_verifycsr( const unsigned char *buf, size_t buflen )
     unsigned char hash[MBEDTLS_MD_MAX_SIZE];
     const mbedtls_md_info_t *md_info;
     mbedtls_x509_csr csr;
+    int ret;
+
+    mbedtls_x509_csr_init( &csr );
 
-    if( mbedtls_x509_csr_parse( &csr, buf, buflen ) != 0 )
-        return( MBEDTLS_ERR_X509_BAD_INPUT_DATA );
+    ret = mbedtls_x509_csr_parse( &csr, buf, buflen );
+    if( ret != 0 )
+        goto exit;
 
     md_info = mbedtls_md_info_from_type( csr.sig_md );
     if( mbedtls_md( md_info, csr.cri.p, csr.cri.len, hash ) != 0 )
     {
         /* Note: this can't happen except after an internal error */
-        return( MBEDTLS_ERR_X509_BAD_INPUT_DATA );
+        ret = MBEDTLS_ERR_X509_BAD_INPUT_DATA;
```

Comment:
Why do we care to return this particular error as opposed to just returning ret as is?

Reply:
I suspect this was copy-pasted from an old version of the X509 module, and I don't think this is right: it could be a hardware failure in an MD accelerator, for example. But here in test code it doesn't really matter, and it's outside the scope of this PR so I preferred to leave it alone.

Reply:
But the call to `mbedtls_pk_verify_ext()` was not left alone, hence why I was asking. Not that it matters much.